[Bug]: market-sim fuzz test panics in SettleMTM

### Problem encountered

Overnight vega-market-sim fuzz testing failing with following panic.

Link to artifacts from run - https://jenkins.vega.rocks/job/common/job/vega-market-sim-reinforcement/118/artifact/network_home/vega-sim-9dz743gl/

```
panic: There's less MTM wins than losses, even accounting for decimals

goroutine 109195 [running]:
go.uber.org/zap/zapcore.CheckWriteAction.OnWrite(0x2?, 0x2?, {0x0?, 0x0?, 0xc00c462960?})
	/jenkins/GOPATH/pkg/mod/go.uber.org/zap@v1.24.0/zapcore/entry.go:198 +0x65
go.uber.org/zap/zapcore.(*CheckedEntry).Write(0xc009b88f70, {0xc005f96680, 0x2, 0x2})
	/jenkins/GOPATH/pkg/mod/go.uber.org/zap@v1.24.0/zapcore/entry.go:264 +0x3ec
go.uber.org/zap.(*Logger).Panic(0xc00c55b7e0?, {0x3c0d05e?, 0xc00c55b900?}, {0xc005f96680, 0x2, 0x2})
	/jenkins/GOPATH/pkg/mod/go.uber.org/zap@v1.24.0/logger.go:258 +0x59
code.vegaprotocol.io/vega/core/settlement.(*Engine).SettleMTM(0xc003c71a00, {0x4294580, 0xc00af53b60}, 0xc00834f640, {0xc002145800, 0x3e, 0x7f6b4b2fd260?})
	/jenkins/workspace/common/vega-market-sim-reinforcement/extern/vega/core/settlement/engine.go:363 +0x181b
code.vegaprotocol.io/vega/core/execution/future.(*Market).confirmMTM(0xc0002b3600, {0x4294580, 0xc00af53b60}, 0x0)
	/jenkins/workspace/common/vega-market-sim-reinforcement/extern/vega/core/execution/future/market.go:2285 +0x207
code.vegaprotocol.io/vega/core/execution/future.(*Market).BlockEnd(0xc0002b3600, {0x4294580?, 0xc00af53b60?})
	/jenkins/workspace/common/vega-market-sim-reinforcement/extern/vega/core/execution/future/market.go:963 +0x48e
code.vegaprotocol.io/vega/core/execution.(*Engine).BlockEnd(0xc0002f4840?, {0x4294580, 0xc00af53b60})
	/jenkins/workspace/common/vega-market-sim-reinforcement/extern/vega/core/execution/engine.go:1474 +0x46
code.vegaprotocol.io/vega/core/processor.(*App).OnEndBlock(0xc000ba1500, 0x7f6b4b2c6a58?)
	/jenkins/workspace/common/vega-market-sim-reinforcement/extern/vega/core/processor/abci.go:924 +0x7c2
code.vegaprotocol.io/vega/core/blockchain/abci.(*App).FinalizeBlock(0xc000a66240, {0xb8?, 0x37ee5a0?}, 0xc00c6b4a80)
	/jenkins/workspace/common/vega-market-sim-reinforcement/extern/vega/core/blockchain/abci/abci.go:183 +0x966
code.vegaprotocol.io/vega/cmd/vega/commands/node.(*appW).FinalizeBlock(0xc00e61ab58?, {0x4294510?, 0xc000126000?}, 0xc00b535e68?)
	/jenkins/workspace/common/vega-market-sim-reinforcement/extern/vega/cmd/vega/commands/node/app_wrapper.go:91 +0x2b
code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).processBlock(0xc00023d9e0)
	/jenkins/workspace/common/vega-market-sim-reinforcement/extern/vega/core/blockchain/nullchain/nullchain.go:198 +0x4c9
code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).ForwardTime(0xc00023d9e0, 0x77359400)
	/jenkins/workspace/common/vega-market-sim-reinforcement/extern/vega/core/blockchain/nullchain/nullchain.go:275 +0x1e5
code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).handleForwardTime.func1()
	/jenkins/workspace/common/vega-market-sim-reinforcement/extern/vega/core/blockchain/nullchain/server.go:132 +0x2a
created by code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).handleForwardTime
	/jenkins/workspace/common/vega-market-sim-reinforcement/extern/vega/core/blockchain/nullchain/server.go:131 +0x24a
```

### Observed behaviour

-

### Expected behaviour

-

### Steps to reproduce

```bash
Either use replay file or reproduce with following steps.

1. cd vega-market-sim
2. poetry shell
3. sh scripts/run-fuzz-test.sh -s 2880
```


### Software version

0b04db862cc0b3d37c85f5f0ce9dd08a04454688

### Failing test

vega-market-sim fuzz test

### Jenkins run

https://jenkins.vega.rocks/blue/organizations/jenkins/common%2Fvega-market-sim-reinforcement/detail/vega-market-sim-reinforcement/118/pipeline

### Configuration used

```shell
{
  "genesis_time": "2023-05-09T10:00:00.000000000Z",
  "chain_id": "CUSTOM",
  "initial_height": "0",
  "consensus_params": {
    "block": {
      "max_bytes": "22020096",
      "max_gas": "-1",
      "time_iota_ms": "1"
    },
    "evidence": {
      "max_age_num_blocks": "100000",
      "max_age_duration": "172800000000000",
      "max_bytes": "1048576"
    },
    "validator": {
      "pub_key_types": ["ed25519"]
    },
    "version": {}
  },
  "validators": [
    {
      "address": "7DFD3744A773C166B36F170E4F5EF3E998703F4E",
      "pub_key": {
        "type": "tendermint/PubKeyEd25519",
        "value": "2lPO6I0p5uCBZzeoA2+ECJ2+57LtLVMjXpxH5za9TvE="
      },
      "power": "10",
      "name": ""
    }
  ],
  "app_hash": "",
  "app_state": {
    "assets": {
      "VOTE": {
        "name": "VOTE",
        "symbol": "VOTE",
        "decimals": 5,
        "min_lp_stake": "1",
        "source": {
          "builtin_asset": {
            "max_faucet_amount_mint": "100000000000"
          }
        }
      }
    },
    "validators": {
      "6NemqXSfN4zfZvAQeIANflEFqQ98izX9C0KLI2dITSY=": {
        "id": "c2b0ab69fb1bd19862c04df401920ea62d3e0bce41a50148240f42017824d17a",
        "vega_pub_key": "dfcf0fa8d67cfa11b3a511e320f3c47807a5e0735cc06dd9c2bc00f7ccc67117",
        "vega_pub_key_index": 1,
        "ethereum_address": "0x9e5BEEC6E56B28cCbd02864840B0f1e0125e42Ce",
        "tm_pub_key": "6NemqXSfN4zfZvAQeIANflEFqQ98izX9C0KLI2dITSY=",
        "info_url": "",
        "country": "",
        "name": "",
        "avatar_url": ""
      }
    },
    "network": {
      "replay_attack_threshold": 150
    },
    "network_parameters": {
      "blockchains.ethereumConfig": "{\"network_id\": \"3\", \"chain_id\": \"3\", \"collateral_bridge_contract\": { \"address\": \"0xa6F1E140daC13002Dfd9789D6dBA59117c717D7a\" }, \"confirmations\": 50, \"staking_bridge_contract\": { \"address\": \"0xfce2CC92203A266a9C8e67461ae5067c78f67235\", \"deployment_block_height\": 11001702}, \"multisig_control_contract\": {\"address\": \"0xCF6d41235911184fe6F35D47207813bFF3B91601\", \"deployment_block_height\": 12710009 } }",
      "governance.proposal.asset.maxClose": "8760h0m0s",
      "governance.proposal.asset.maxEnact": "8760h0m0s",
      "governance.proposal.asset.minClose": "1s",
      "governance.proposal.asset.minEnact": "1s",
      "governance.proposal.asset.minProposerBalance": "1",
      "governance.proposal.asset.minVoterBalance": "1",
      "governance.proposal.asset.requiredMajority": "0.66",
      "governance.proposal.asset.requiredParticipation": "0.00001",
      "governance.proposal.freeform.maxClose": "8760h0m0s",
      "governance.proposal.freeform.minClose": "1s",
      "governance.proposal.freeform.minProposerBalance": "1",
      "governance.proposal.freeform.minVoterBalance": "1",
      "governance.proposal.freeform.requiredMajority": "0.66",
      "governance.proposal.freeform.requiredParticipation": "0.00001",
      "governance.proposal.market.maxClose": "8760h0m0s",
      "governance.proposal.market.maxEnact": "8760h0m0s",
      "governance.proposal.market.minClose": "1s",
      "governance.proposal.market.minEnact": "2s",
      "governance.proposal.market.minProposerBalance": "1",
      "governance.proposal.market.minVoterBalance": "1",
      "governance.proposal.market.requiredMajority": "0.66",
      "governance.proposal.market.requiredParticipation": "0.00001",
      "governance.proposal.updateAsset.maxClose": "720h",
      "governance.proposal.updateAsset.maxEnact": "720h",
      "governance.proposal.updateAsset.minClose": "1s",
      "governance.proposal.updateAsset.minEnact": "1s",
      "governance.proposal.updateAsset.minProposerBalance": "1",
      "governance.proposal.updateAsset.minVoterBalance": "1",
      "governance.proposal.updateAsset.requiredMajority": "0.66",
      "governance.proposal.updateAsset.requiredParticipation": "0.09",
      "governance.proposal.updateMarket.maxClose": "8760h0m0s",
      "governance.proposal.updateMarket.maxEnact": "8760h0m0s",
      "governance.proposal.updateMarket.minClose": "1s",
      "governance.proposal.updateMarket.minEnact": "1s",
      "governance.proposal.updateMarket.minProposerBalance": "1",
      "governance.proposal.updateMarket.minProposerEquityLikeShare": "0",
      "governance.proposal.updateMarket.minVoterBalance": "1",
      "governance.proposal.updateMarket.requiredMajority": "0.66",
      "governance.proposal.updateMarket.requiredMajorityLP": "0.66",
      "governance.proposal.updateMarket.requiredParticipation": "0.00001",
      "governance.proposal.updateMarket.requiredParticipationLP": "0",
      "governance.proposal.updateNetParam.maxClose": "8760h0m0s",
      "governance.proposal.updateNetParam.maxEnact": "8760h0m0s",
      "governance.proposal.updateNetParam.minClose": "1s",
      "governance.proposal.updateNetParam.minEnact": "1s",
      "governance.proposal.updateNetParam.minProposerBalance": "1",
      "governance.proposal.updateNetParam.minVoterBalance": "1",
      "governance.proposal.updateNetParam.requiredMajority": "0.5",
      "governance.proposal.updateNetParam.requiredParticipation": "0.00001",
      "governance.proposal.referralProgram.maxClose": "8760h0m0s",
      "governance.proposal.referralProgram.maxEnact": "8760h0m0s",
      "governance.proposal.referralProgram.minClose": "1s",
      "governance.proposal.referralProgram.minEnact": "1s",
      "governance.proposal.referralProgram.minProposerBalance": "1",
      "governance.proposal.referralProgram.minVoterBalance": "1",
      "governance.proposal.referralProgram.requiredMajority": "0.5",
      "governance.proposal.referralProgram.requiredParticipation": "0.00001",
      "governance.proposal.VolumeDiscountProgram.maxClose": "8760h0m0s",
      "governance.proposal.VolumeDiscountProgram.maxEnact": "8760h0m0s",
      "governance.proposal.VolumeDiscountProgram.minClose": "1s",
      "governance.proposal.VolumeDiscountProgram.minEnact": "1s",
      "governance.proposal.VolumeDiscountProgram.minProposerBalance": "1",
      "governance.proposal.VolumeDiscountProgram.minVoterBalance": "1",
      "governance.proposal.VolumeDiscountProgram.requiredMajority": "0.5",
      "governance.proposal.VolumeDiscountProgram.requiredParticipation": "0.00001",
      "governance.proposal.transfer.maxClose": "8760h0m0s",
      "governance.proposal.transfer.maxEnact": "8760h0m0s",
      "governance.proposal.transfer.minClose": "1s",
      "governance.proposal.transfer.minEnact": "1s",
      "governance.proposal.transfer.minProposerBalance": "1",
      "governance.proposal.transfer.minVoterBalance": "1",
      "governance.proposal.transfer.requiredMajority": "0.5",
      "governance.proposal.transfer.requiredParticipation": "0.00001",
      "limits.markets.maxPeggedOrders": "1500",
      "limits.markets.proposePerpetualEnabled": "1",
      "market.auction.maximumDuration": "168h",
      "market.auction.minimumDuration": "1s",
      "market.fee.factors.infrastructureFee": "0.0005",
      "market.fee.factors.makerFee": "0.0002",
      "market.liquidity.bondPenaltyParameter": "0.1",
      "market.liquidity.earlyExitPenalty": "0.1",
      "market.liquidity.maximumLiquidityFeeFactorLevel": "0.03",
      "market.liquidity.minimum.probabilityOfTrading.lpOrders": "1e-6",
      "market.liquidity.probabilityOfTrading.tau.scaling": "10.0",
      "market.liquidity.providersFeeCalculationTimeStep": "1s",
      "market.liquidity.sla.nonPerformanceBondPenaltyMax": "0.5",
      "market.liquidity.sla.nonPerformanceBondPenaltySlope": "2",
      "market.liquidity.stakeToCcyVolume": "1.0",
      "market.liquidity.targetstake.triggering.ratio": "0.25",
      "market.liquidityProvision.minLpStakeQuantumMultiple": "5000",
      "market.liquidityProvision.shapes.maxSize": "100",
      "market.margin.scalingFactors": "{\"search_level\": 1.1, \"initial_margin\": 1.5, \"collateral_release\": 1.7}",
      "market.monitor.price.defaultParameters": "{\"triggers\": [{\"auction_extension\": 300, \"horizon\": 43200, \"probability\": \"0.9999999\"}] }",
      "market.stake.target.scalingFactor": "0.0001",
      "market.stake.target.timeWindow": "1h",
      "market.value.windowLength": "2h0m0s",
      "network.checkpoint.timeElapsedBetweenCheckpoints": "5m",
      "network.floatingPointUpdates.delay": "5m",
      "network.markPriceUpdateMaximumFrequency": "5s",
      "network.transaction.defaultgas": "1",
      "network.transactions.maxgasperblock": "3000000",
      "network.transactions.minBlockCapacity": "32",
      "network.validators.ersatz.multipleOfTendermintValidators": "0",
      "network.validators.ersatz.rewardFactor": "0.85",
      "network.validators.incumbentBonus": "0.05",
      "network.validators.minimumEthereumEventsForNewValidator": "3",
      "network.validators.multisig.numberOfSigners": "13",
      "network.validators.tendermint.number": "13",
      "reward.asset": "VOTE",
      "reward.staking.delegation.competitionLevel": "1.3",
      "reward.staking.delegation.delegatorShare": "0.8",
      "reward.staking.delegation.maxPayoutPerEpoch": "9890000000000000000000",
      "reward.staking.delegation.maxPayoutPerParticipant": "700000000000000000000",
      "reward.staking.delegation.minimumValidatorStake": "3000000000000000000000",
      "reward.staking.delegation.minValidators": "5",
      "reward.staking.delegation.optimalStakeMultiplier": "3.0",
      "reward.staking.delegation.payoutDelay": "0h",
      "reward.staking.delegation.payoutFraction": "1.0",
      "rewards.marketCreationQuantumMultiple": "1",
      "snapshot.interval.length": "300",
      "spam.pow.difficulty": "15",
      "spam.pow.hashFunction": "sha3_24_rounds",
      "spam.pow.increaseDifficulty": "0",
      "spam.pow.numberOfPastBlocks": "100",
      "spam.pow.numberOfTxPerBlock": "1000",
      "spam.protection.delegation.min.tokens": "100000000000000000",
      "spam.protection.max.batchSize": "30",
      "spam.protection.max.delegations": "360",
      "spam.protection.max.proposals": "300",
      "spam.protection.max.votes": "300",
      "spam.protection.maxUserTransfersPerEpoch": "10000",
      "spam.protection.minimumWithdrawalQuantumMultiple": "10",
      "spam.protection.minMultisigUpdates": "100000000000000000000",
      "spam.protection.proposal.min.tokens": "2000000000000000000000",
      "spam.protection.voting.min.tokens": "1000000000000000000",
      "spam.protection.applyReferral.min.funds": "0",
      "transfer.fee.factor": "0.001",
      "transfer.minTransferQuantumMultiple": "100",
      "validator.performance.scaling.factor": "0",
      "validators.delegation.minAmount": "1",
      "validators.epoch.length": "2m",
      "validators.vote.required": "0.67",
      "referralProgram.maxReferralTiers": "10",
      "referralProgram.maxReferralRewardFactor": "1",
      "referralProgram.maxReferralDiscountFactor": "1",
      "referralProgram.maxPartyNotionalVolumeByQuantumPerEpoch": "10000",
      "referralProgram.minStakedVegaTokens": "0",
      "referralProgram.maxReferralRewardProportion": "1"
    },
    "network_limits": {
      "propose_market_enabled": true,
      "propose_asset_enabled": true,
      "bootstrap_block_count": 0
    },
    "checkpoint": {
      "load_hash": ""
    }
  }
}
```


### Relevant log output

```shell
panic: There's less MTM wins than losses, even accounting for decimals

goroutine 109195 [running]:
go.uber.org/zap/zapcore.CheckWriteAction.OnWrite(0x2?, 0x2?, {0x0?, 0x0?, 0xc00c462960?})
	/jenkins/GOPATH/pkg/mod/go.uber.org/zap@v1.24.0/zapcore/entry.go:198 +0x65
go.uber.org/zap/zapcore.(*CheckedEntry).Write(0xc009b88f70, {0xc005f96680, 0x2, 0x2})
	/jenkins/GOPATH/pkg/mod/go.uber.org/zap@v1.24.0/zapcore/entry.go:264 +0x3ec
go.uber.org/zap.(*Logger).Panic(0xc00c55b7e0?, {0x3c0d05e?, 0xc00c55b900?}, {0xc005f96680, 0x2, 0x2})
	/jenkins/GOPATH/pkg/mod/go.uber.org/zap@v1.24.0/logger.go:258 +0x59
code.vegaprotocol.io/vega/core/settlement.(*Engine).SettleMTM(0xc003c71a00, {0x4294580, 0xc00af53b60}, 0xc00834f640, {0xc002145800, 0x3e, 0x7f6b4b2fd260?})
	/jenkins/workspace/common/vega-market-sim-reinforcement/extern/vega/core/settlement/engine.go:363 +0x181b
code.vegaprotocol.io/vega/core/execution/future.(*Market).confirmMTM(0xc0002b3600, {0x4294580, 0xc00af53b60}, 0x0)
	/jenkins/workspace/common/vega-market-sim-reinforcement/extern/vega/core/execution/future/market.go:2285 +0x207
code.vegaprotocol.io/vega/core/execution/future.(*Market).BlockEnd(0xc0002b3600, {0x4294580?, 0xc00af53b60?})
	/jenkins/workspace/common/vega-market-sim-reinforcement/extern/vega/core/execution/future/market.go:963 +0x48e
code.vegaprotocol.io/vega/core/execution.(*Engine).BlockEnd(0xc0002f4840?, {0x4294580, 0xc00af53b60})
	/jenkins/workspace/common/vega-market-sim-reinforcement/extern/vega/core/execution/engine.go:1474 +0x46
code.vegaprotocol.io/vega/core/processor.(*App).OnEndBlock(0xc000ba1500, 0x7f6b4b2c6a58?)
	/jenkins/workspace/common/vega-market-sim-reinforcement/extern/vega/core/processor/abci.go:924 +0x7c2
code.vegaprotocol.io/vega/core/blockchain/abci.(*App).FinalizeBlock(0xc000a66240, {0xb8?, 0x37ee5a0?}, 0xc00c6b4a80)
	/jenkins/workspace/common/vega-market-sim-reinforcement/extern/vega/core/blockchain/abci/abci.go:183 +0x966
code.vegaprotocol.io/vega/cmd/vega/commands/node.(*appW).FinalizeBlock(0xc00e61ab58?, {0x4294510?, 0xc000126000?}, 0xc00b535e68?)
	/jenkins/workspace/common/vega-market-sim-reinforcement/extern/vega/cmd/vega/commands/node/app_wrapper.go:91 +0x2b
code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).processBlock(0xc00023d9e0)
	/jenkins/workspace/common/vega-market-sim-reinforcement/extern/vega/core/blockchain/nullchain/nullchain.go:198 +0x4c9
code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).ForwardTime(0xc00023d9e0, 0x77359400)
	/jenkins/workspace/common/vega-market-sim-reinforcement/extern/vega/core/blockchain/nullchain/nullchain.go:275 +0x1e5
code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).handleForwardTime.func1()
	/jenkins/workspace/common/vega-market-sim-reinforcement/extern/vega/core/blockchain/nullchain/server.go:132 +0x2a
created by code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).handleForwardTime
	/jenkins/workspace/common/vega-market-sim-reinforcement/extern/vega/core/blockchain/nullchain/server.go:131 +0x24a
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -51,6 +51,7 @@
 - [10127](https://github.com/vegaprotocol/vega/issues/10127) - Untangle `ApplyReferralCode` and `JoinTeam` command verification.
 - [10153](https://github.com/vegaprotocol/vega/issues/10153) - Add metrics and reduce amount of request sent to the Ethereum `RPC`.
 - [10147](https://github.com/vegaprotocol/vega/issues/10147) - Add network transfer largest share to the transfers if needed.
+- [10158](https://github.com/vegaprotocol/vega/issues/10158) - Add the network as the zero-share default party in settlement engine.
 
 ## 0.73.0
 

--- a/core/settlement/engine.go
+++ b/core/settlement/engine.go
@@ -152,7 +152,7 @@ func (e *Engine) Settle(t time.Time, settlementData *num.Uint) ([]*types.Transfe
 // each change in position has to be calculated using the exact price of the trade.
 func (e *Engine) AddTrade(trade *types.Trade) {
 	e.mu.Lock()
-	defer d.mu.Unlock()
+	defer e.mu.Unlock()
 	// network registers a wash trade to update its position
 	if trade.Buyer == types.NetworkParty && trade.Buyer == trade.Seller {
 		e.addNetworkTrade(trade)
@@ -198,32 +198,19 @@ func (e *Engine) AddTrade(trade *types.Trade) {
 }
 
 func (e *Engine) addNetworkTrade(trade *types.Trade) {
-	var size, tSize int64
-	if trade.Aggressor == types.SideBuy {
-		if cd, ok := e.trades[trade.Buyer]; !ok || len(cd) == 0 {
-			e.trades[trade.Buyer] = []*settlementTrade{}
-			// check if the buyer already has a known position
-			if pos, ok := e.settledPosition[trade.Buyer]; ok {
-				size = pos
-			}
-		} else {
-			size = cd[len(cd)-1].newSize
-		}
-		tSize = int64(trade.Size)
-	} else {
-		if cd, ok := e.trades[trade.Seller]; !ok || len(cd) == 0 {
-			e.trades[trade.Seller] = []*settlementTrade{}
-			// check if seller has a known position
-			if pos, ok := e.settledPosition[trade.Seller]; ok {
-				size = pos
-			}
-		} else {
-			size = cd[len(cd)-1].newSize
-		}
-		tSize = -int64(trade.Size)
+	tSize := int64(trade.Size)
+	// sell wash trade, the settled position of the network decreases.
+	if trade.Aggressor == types.SideSell {
+		tSize *= -1
 	}
-	size += tSize
-	e.settledPosition[types.NetworkParty] = size
+	e.settledPosition[types.NetworkParty] += tSize
+	trades := e.trades[types.NetworkParty]
+	// all trades should be based on the new settled position size
+	for i, t := range trades {
+		t.newSize += tSize
+		trades[i] = t
+	}
+	e.trades[types.NetworkParty] = trades
 }
 
 func (e *Engine) HasTraded() bool {

--- a/core/settlement/engine.go
+++ b/core/settlement/engine.go
@@ -387,10 +387,12 @@ func (e *Engine) SettleMTM(ctx context.Context, markPrice *num.Uint, positions [
 	}
 	// append wins after loss transfers
 	transfers = append(transfers, wins...)
-	if len(transfers) > 0 && appendLargest {
+	if len(transfers) > 0 && appendLargest && largestShare.transfer != nil {
 		transfers = append(transfers, largestShare)
 	}
-	e.broker.SendBatch(evts)
+	if len(evts) > 0 {
+		e.broker.SendBatch(evts)
+	}
 	timer.EngineTimeCounterAdd()
 	return transfers
 }

--- a/core/settlement/engine.go
+++ b/core/settlement/engine.go
@@ -152,6 +152,12 @@ func (e *Engine) Settle(t time.Time, settlementData *num.Uint) ([]*types.Transfe
 // each change in position has to be calculated using the exact price of the trade.
 func (e *Engine) AddTrade(trade *types.Trade) {
 	e.mu.Lock()
+	// network registers a wash trade to update its position
+	if trade.Buyer == types.NetworkParty && trade.Buyer == trade.Seller {
+		e.addNetworkTrade(trade)
+		e.mu.Unlock()
+		return
+	}
 	var buyerSize, sellerSize int64
 	// checking the len of cd shouldn't be required here, but it is needed in the second if
 	// in case the buyer and seller are one and the same...
@@ -190,6 +196,35 @@ func (e *Engine) AddTrade(trade *types.Trade) {
 		newSize:     sellerSize - size,
 	})
 	e.mu.Unlock()
+}
+
+func (e *Engine) addNetworkTrade(trade *types.Trade) {
+	var size, tSize int64
+	if trade.Aggressor == types.SideBuy {
+		if cd, ok := e.trades[trade.Buyer]; !ok || len(cd) == 0 {
+			e.trades[trade.Buyer] = []*settlementTrade{}
+			// check if the buyer already has a known position
+			if pos, ok := e.settledPosition[trade.Buyer]; ok {
+				size = pos
+			}
+		} else {
+			size = cd[len(cd)-1].newSize
+		}
+		tSize = int64(trade.Size)
+	} else {
+		if cd, ok := e.trades[trade.Seller]; !ok || len(cd) == 0 {
+			e.trades[trade.Seller] = []*settlementTrade{}
+			// check if seller has a known position
+			if pos, ok := e.settledPosition[trade.Seller]; ok {
+				size = pos
+			}
+		} else {
+			size = cd[len(cd)-1].newSize
+		}
+		tSize = -int64(trade.Size)
+	}
+	size += tSize
+	e.settledPosition[types.NetworkParty] = size
 }
 
 func (e *Engine) HasTraded() bool {

--- a/core/settlement/engine.go
+++ b/core/settlement/engine.go
@@ -356,6 +356,9 @@ func (e *Engine) SettleMTM(ctx context.Context, markPrice *num.Uint, positions [
 	delta := num.UintZero().Sub(lossTotal, winTotal)
 	if !delta.IsZero() {
 		if zeroAmts {
+			if appendLargest {
+				zeroShares = append(zeroShares, largestShare)
+			}
 			zRound := num.DecimalFromInt64(int64(len(zeroShares)))
 			// there are more transfers from losses than we pay out to wins, but some winning parties have zero transfers
 			// this delta should == combined win decimals, let's sanity check this!

--- a/core/settlement/engine.go
+++ b/core/settlement/engine.go
@@ -152,10 +152,10 @@ func (e *Engine) Settle(t time.Time, settlementData *num.Uint) ([]*types.Transfe
 // each change in position has to be calculated using the exact price of the trade.
 func (e *Engine) AddTrade(trade *types.Trade) {
 	e.mu.Lock()
+	defer d.mu.Unlock()
 	// network registers a wash trade to update its position
 	if trade.Buyer == types.NetworkParty && trade.Buyer == trade.Seller {
 		e.addNetworkTrade(trade)
-		e.mu.Unlock()
 		return
 	}
 	var buyerSize, sellerSize int64
@@ -195,7 +195,6 @@ func (e *Engine) AddTrade(trade *types.Trade) {
 		size:        -size,
 		newSize:     sellerSize - size,
 	})
-	e.mu.Unlock()
 }
 
 func (e *Engine) addNetworkTrade(trade *types.Trade) {

--- a/core/settlement/engine_test.go
+++ b/core/settlement/engine_test.go
@@ -752,8 +752,297 @@ func testAddNewPartySelfTrade(t *testing.T) {
 	engine.Update(init)
 	engine.AddTrade(trade)
 	noTransfers := engine.SettleMTM(context.Background(), markPrice, positions)
-	assert.Len(t, noTransfers, 2)
+	assert.Len(t, noTransfers, 1)
 	assert.Nil(t, noTransfers[0].Transfer())
+}
+
+func TestNetworkPartyCloseout(t *testing.T) {
+	engine := getTestEngine(t)
+	currentMP := num.NewUint(1000)
+	// network trade
+	nTrade := &types.Trade{
+		Buyer:  types.NetworkParty,
+		Seller: types.NetworkParty,
+		Price:  currentMP.Clone(),
+		Size:   10,
+	}
+	nPosition := testPos{
+		party: types.NetworkParty,
+		size:  10,
+		price: currentMP.Clone(),
+	}
+	sellPrice := num.NewUint(990)
+	// now trade with health party, at some different price
+	// trigger a loss for the network.
+	cTrade := &types.Trade{
+		Buyer:  "party1",
+		Seller: types.NetworkParty,
+		Size:   2,
+		Price:  sellPrice.Clone(),
+	}
+	init := []events.MarketPosition{
+		nPosition,
+		testPos{
+			party: "party1",
+			size:  0,
+			price: currentMP.Clone(),
+		},
+	}
+	positions := []events.MarketPosition{
+		testPos{
+			party: types.NetworkParty,
+			size:  8,
+			price: currentMP.Clone(),
+		},
+		testPos{
+			party: "party1",
+			size:  2,
+			price: currentMP.Clone(),
+		},
+	}
+	engine.Update(init)
+	engine.AddTrade(nTrade)
+	engine.AddTrade(cTrade)
+	transfers := engine.SettleMTM(context.Background(), currentMP, positions)
+	assert.NotEmpty(t, transfers)
+	assert.Len(t, transfers, 2)
+}
+
+func TestNetworkCloseoutZero(t *testing.T) {
+	engine := getTestEngine(t)
+	currentMP := num.NewUint(1000)
+	// network trade
+	nTrade := &types.Trade{
+		Buyer:  types.NetworkParty,
+		Seller: types.NetworkParty,
+		Price:  currentMP.Clone(),
+		Size:   10,
+	}
+	nPosition := testPos{
+		party: types.NetworkParty,
+		size:  10,
+		price: currentMP.Clone(),
+	}
+	sellPrice := num.NewUint(999)
+	// now trade with health party, at some different price
+	// trigger a loss for the network.
+	cTrades := []*types.Trade{
+		{
+			Buyer:  "party1",
+			Seller: types.NetworkParty,
+			Size:   1,
+			Price:  sellPrice.Clone(),
+		},
+		{
+			Buyer:  "party2",
+			Seller: types.NetworkParty,
+			Size:   1,
+			Price:  sellPrice.Clone(),
+		},
+	}
+	init := []events.MarketPosition{
+		nPosition,
+		testPos{
+			party: "party1",
+			size:  0,
+			price: currentMP.Clone(),
+		},
+		testPos{
+			party: "party2",
+			size:  0,
+			price: currentMP.Clone(),
+		},
+		testPos{
+			party: "party3",
+			size:  -5,
+			price: currentMP.Clone(),
+		},
+		testPos{
+			party: "party4",
+			size:  -5,
+			price: currentMP.Clone(),
+		},
+	}
+	positions := []events.MarketPosition{
+		testPos{
+			party: types.NetworkParty,
+			size:  8,
+			price: currentMP.Clone(),
+		},
+		testPos{
+			party: "party1",
+			size:  1,
+			price: currentMP.Clone(),
+		},
+		testPos{
+			party: "party2",
+			size:  1,
+			price: currentMP.Clone(),
+		},
+		testPos{
+			party: "party3",
+			size:  -5,
+			price: currentMP.Clone(),
+		},
+		testPos{
+			party: "party4",
+			size:  -5,
+			price: currentMP.Clone(),
+		},
+	}
+	engine.Update(init)
+	engine.AddTrade(nTrade)
+	for _, cTrade := range cTrades {
+		engine.AddTrade(cTrade)
+	}
+	transfers := engine.SettleMTM(context.Background(), currentMP, positions)
+	assert.NotEmpty(t, transfers)
+	// now that the network has an established long position, make short positions close out and mark to market
+	// party 3 closes their position, lowering the mark price
+	newMP := num.NewUint(990)
+	trade := &types.Trade{
+		Buyer:  "party3",
+		Seller: "party1",
+		Price:  newMP,
+		Size:   1,
+	}
+	positions = []events.MarketPosition{
+		testPos{
+			party: types.NetworkParty,
+			size:  8,
+			price: newMP.Clone(),
+		},
+		testPos{
+			party: "party1",
+			size:  0,
+			price: newMP.Clone(),
+		},
+		testPos{
+			party: "party2",
+			size:  1,
+			price: newMP.Clone(),
+		},
+		testPos{
+			party: "party3",
+			size:  -4,
+			price: newMP.Clone(),
+		},
+		testPos{
+			party: "party4",
+			size:  -5,
+			price: newMP.Clone(),
+		},
+	}
+	engine.AddTrade(trade)
+	transfers = engine.SettleMTM(context.Background(), newMP, positions)
+	assert.NotEmpty(t, transfers)
+	// now make it look like party2 got distressed because of this MTM settlement
+	nTrade = &types.Trade{
+		Price:  newMP.Clone(),
+		Size:   1,
+		Buyer:  types.NetworkParty,
+		Seller: types.NetworkParty,
+	}
+	positions = []events.MarketPosition{
+		testPos{
+			party: types.NetworkParty,
+			size:  9,
+			price: newMP.Clone(),
+		},
+		testPos{
+			party: "party3",
+			size:  -4,
+			price: newMP.Clone(),
+		},
+		testPos{
+			party: "party4",
+			size:  -5,
+			price: newMP.Clone(),
+		},
+	}
+	engine.AddTrade(nTrade)
+	transfers = engine.SettleMTM(context.Background(), newMP, positions) // amounts are zero here (was trade only)
+	assert.NotEmpty(t, transfers)
+	newMP = num.NewUint(995)
+	positions = []events.MarketPosition{
+		testPos{
+			party: types.NetworkParty,
+			size:  9,
+			price: newMP.Clone(),
+		},
+		testPos{
+			party: "party3",
+			size:  -4,
+			price: newMP.Clone(),
+		},
+		testPos{
+			party: "party4",
+			size:  -5,
+			price: newMP.Clone(),
+		},
+	}
+	transfers = engine.SettleMTM(context.Background(), newMP.Clone(), positions)
+	assert.NotEmpty(t, transfers)
+	// now the same, but network loses
+	newMP = num.NewUint(990)
+	positions = []events.MarketPosition{
+		testPos{
+			party: types.NetworkParty,
+			size:  9,
+			price: newMP.Clone(),
+		},
+		testPos{
+			party: "party3",
+			size:  -4,
+			price: newMP.Clone(),
+		},
+		testPos{
+			party: "party4",
+			size:  -5,
+			price: newMP.Clone(),
+		},
+	}
+	transfers = engine.SettleMTM(context.Background(), newMP.Clone(), positions)
+	assert.NotEmpty(t, transfers)
+	// assume no trades occurred, but the mark price has changed (shouldn't happen, but this could end up with a situation where network profits without trading)
+	// network disposes of its position and profits
+	disposePrice := num.NewUint(1010)
+	trades := []*types.Trade{
+		{
+			Seller: types.NetworkParty,
+			Buyer:  "party3",
+			Price:  disposePrice.Clone(),
+			Size:   4,
+		},
+		{
+			Seller: types.NetworkParty,
+			Buyer:  "party4",
+			Price:  disposePrice.Clone(),
+			Size:   5,
+		},
+	}
+	positions = []events.MarketPosition{
+		testPos{
+			party: types.NetworkParty,
+			size:  0,
+			price: newMP.Clone(),
+		},
+		testPos{
+			party: "party3",
+			size:  0,
+			price: newMP.Clone(),
+		},
+		testPos{
+			party: "party4",
+			size:  0,
+			price: newMP.Clone(),
+		},
+	}
+	for _, tr := range trades {
+		engine.AddTrade(tr)
+	}
+	transfers = engine.SettleMTM(context.Background(), num.NewUint(1000), positions)
+	assert.NotEmpty(t, transfers)
 }
 
 func testAddNewParty(t *testing.T) {
@@ -796,7 +1085,7 @@ func testAddNewParty(t *testing.T) {
 	engine.Update(init)
 	engine.AddTrade(trade)
 	noTransfers := engine.SettleMTM(context.Background(), markPrice, positions)
-	assert.Len(t, noTransfers, 3)
+	assert.Len(t, noTransfers, 2)
 	for _, v := range noTransfers {
 		assert.Nil(t, v.Transfer())
 	}

